modules/deployment-scripts/build-acr/main.bicep

```diff
@@ -0,0 +1,140 @@
+param gitRepositoryUrl string
+
+@description('The name of the repository branch to use')
+param gitBranch string = 'main'
+
+@description('The directory in the repo that contains the dockerfile')
+param gitRepoDirectory string = ''
+
+@description('The image name/path you want to create in ACR')
+param imageName string
+
+@description('The image tag you want to create')
+param imageTag string = string(dateTimeToEpoch(utcNow()))
+
+@description('The ACR compute platform needed to build the image')
+param acrBuildPlatform string = 'linux'
+
+var repo = '${gitRepositoryUrl}#${gitBranch}:${gitRepoDirectory}'
+var cleanRepoName = last(split(gitRepositoryUrl, '/'))
+var tag = '${imageName}:${imageTag}'
+
+resource acr 'Microsoft.ContainerRegistry/registries@2021-12-01-preview' existing = {
+  name: AcrName
+}
+
+resource newDepScriptId 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = if (!useExistingManagedIdentity) {
+  name: managedIdentityName
+  location: location
+}
+
+resource existingDepScriptId 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' existing = if (useExistingManagedIdentity ) {
+  name: managedIdentityName
+  scope: resourceGroup(existingManagedIdentitySubId, existingManagedIdentityResourceGroupName)
+}
+
+resource rbac 'Microsoft.Authorization/roleAssignments@2022-04-01' = if (!empty(rbacRoleNeeded)) {
+  name: guid(acr.id, rbacRoleNeeded, useExistingManagedIdentity ? existingDepScriptId.id : newDepScriptId.id)
+  scope: acr
+  properties: {
+    roleDefinitionId: resourceId('Microsoft.Authorization/roleDefinitions', rbacRoleNeeded)
+    principalId: useExistingManagedIdentity ? existingDepScriptId.properties.principalId : newDepScriptId.properties.principalId
+    principalType: 'ServicePrincipal'
+  }
+}
+
+resource createImportImage 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
+  name: 'ACR-Build-${cleanRepoName}'
+  location: location
+  identity: {
+    type: 'UserAssigned'
+    userAssignedIdentities: {
+      '${useExistingManagedIdentity ? existingDepScriptId.id : newDepScriptId.id}': {}
+    }
+  }
+  kind: 'AzureCLI'
+  dependsOn: [
+    rbac
+  ]
+  properties: {
+    forceUpdateTag: forceUpdateTag
+    azCliVersion: '2.30.0'
+    timeout: 'PT45M'
+    retentionInterval: 'P1D'
+    environmentVariables: [
+      {
+        name: 'acrName'
+        value: acr.name
+      }
+      {
+        name: 'acrResourceGroup'
+        value: resourceGroup().name
+      }
+      {
+        name: 'tag'
+        value: tag
+      }
+      {
+        name: 'repo'
+        value: repo
+      }
+      {
+        name: 'platform'
+        value: acrBuildPlatform
+      }
+      {
+        name: 'initialDelay'
+        value: initialScriptDelay
+      }
+    ]
+    scriptContent: '''
+      #!/bin/bash
+      set -e
+      
+      echo "Waiting on RBAC replication ($initialDelay)"
+      sleep $initialDelay
+      
+      az acr build -g $acrResourceGroup -r $acrName -t $tag $repo --platform $platform
```

Comment:
```suggestion
      az acr build \
        --resource-group $acrResourceGroup \
        --registry $acrName \
        --image $tag $repo \
        --platform $platform

```

Reply:
nitpick: minor formatting suggestion which I think makes it a little easier to read. Also prefer to use full names for parameters to make it a little easier for someone reading it who might not know the command.

Reply:
this script is short enough that it's probably fine the way it is, but I also like to use `loadTextContent()`, to load a script from its own file. Having the shell script in its own file is nice because it lights up highlighting and formatting for the file type. Also like to use Shellcheck to check over the scripts (have a github action for that somewhere)

Reply:
> this script is short enough that it's probably fine the way it is, but I also like to use `loadTextContent()`, to load a script from its own file. Having the shell script in its own file is nice because it lights up highlighting and formatting for the file type. Also like to use Shellcheck to check over the scripts (have a github action for that somewhere)

I like this, will do another pr to update 😎

Reply:
I've put your suggestion into this PR @dciborow  👍